webapp/src/components/common/fieldEditors/NewSelectFE.tsx

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright (c) 2025, RTE (https://www.rte-france.com)
+ *
+ * See AUTHORS.txt
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This file is part of the Antares project.
+ */
+
+import reactHookFormSupport, { type ReactHookFormSupportProps } from "@/hoc/reactHookFormSupport";
+import type { SvgIconComponent } from "@mui/icons-material";
+import {
+  Box,
+  Button,
+  ButtonGroup,
+  Chip,
+  MenuItem,
+  TextField,
+  Tooltip,
+  type TextFieldProps,
+} from "@mui/material";
+import startCase from "lodash/startCase";
+import * as RA from "ramda-adjunct";
+import { useMemo } from "react";
+import type { FieldPath, FieldValues } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+
+type Value = string | number;
+
+interface Option<T extends Value> {
+  label: string;
+  value: T;
+  icon?: SvgIconComponent;
+  tooltip?: string;
+}
+
+type Options<T extends Value> = Array<T | Option<T>>;
+
+interface SingleSelect {
+  multiple?: false;
+  value?: Value;
+  defaultValue?: Value;
+  onSelectAllOptions?: never;
+  onDeselectAllOptions?: never;
+}
+
+interface MultipleSelect<OptionValue extends Value = Value> {
+  multiple: true;
+  // `value` or `defaultValue` may differ from `options` at initialization.
+  // Reason why `Value` is used instead of `OptionValue`.
+  value?: Value[];
+  defaultValue?: Value[];
+  onSelectAllOptions?: (optionValues: OptionValue[]) => void;
+  onDeselectAllOptions?: VoidFunction;
+}
+
+type NewSelectFEProps<OptionValue extends Value = Value> = {
+  options?: Options<OptionValue>;
+  emptyValue?: boolean;
+  startCaseLabel?: boolean;
+  renderValueAs?: "text" | "chip";
```

Comment:
you could remove the `renderValueAs` prop entirely and always use chips for multiple selects (since that's the most intuitive UI pattern)

```ts
    slotProps={{
        ...slotProps,
        select: {
          renderValue: multiple ? renderSelectedValueAsChip : undefined,
          multiple,
        },
      }}
```